main.go

```diff
@@ -127,22 +127,39 @@ func (n *netcupDNSProviderSolver) getConfig(ch *v1alpha1.ChallengeRequest) (*net
 	return netcupCfg, nil
 }
 
-func addOrDeleteTxtRecord(cfg *netcupClientConfig, resolvedFqdn string, dnsName string, key string, delete bool) error {
+func reverseArray(arr []string) []string {
+	for i, j := 0, len(arr)-1; i < j; i, j = i+1, j-1 {
+		arr[i], arr[j] = arr[j], arr[i]
+	}
+	return arr
+}
+
+func addOrDeleteTxtRecord(cfg *netcupClientConfig, resolvedFqdn string, key string, delete bool) error {
 	netcupClient := netcup.NewNetcupDnsClient(cfg.customerNumber, cfg.apiKey, cfg.apiPassword)
 	sess, err := netcupClient.Login()
 	if err != nil {
 		return fmt.Errorf("unable to login to netcup API: %v", err)
 	}
 	defer sess.Logout()
 
-	recs, err := sess.InfoDnsRecords(dnsName)
+	splitDomainName := strings.Split(strings.TrimSuffix(resolvedFqdn, "."), ".")
+	host := ""
+	if len(splitDomainName) == 4 {
```

Comment:
Regarding this logic here: wouldn't it be better to use regex again and capture last two segments to a domain name + TLD (apparently not needed elsewhere) and capture _everything else_ before it into ``host``? In my opinion it would make the code much easier to understand and also fit your use case? What i am thinking on may be tested here: https://go.dev/play/p/9MRWe_nBcIH

Reply:
What i also just found out, that there is a ``ResolvedZone`` in challenge request, maybe this can be of use, too.

Reply:
@DerKnerd i've implemented the alternative approach in [this commit](https://github.com/aellwein/cert-manager-webhook-netcup/commit/03c4303168386bedcbf4cc3fe926bf72e35cdbc7). From my testing, it behaves about the same as in your PR. The difference is, it can also parse longer FQDNs (e.g. more than 4 segments in DNS name). Tell me what you think about it.